src/payment/models/crypto-route/crypto-route.controller.ts

```diff
@@ -54,6 +57,13 @@ export class CryptoRouteController {
     return this.cryptoRouteService.updateCrypto(jwt.id, +id, updateCryptoDto).then((b) => this.toDto(jwt.id, b));
   }
 
+  @Get(':id/history')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  async getCryptoRouteHistory(@GetJwt() jwt: JwtPayload, @Param('id') id: string): Promise<CryptoInputHistoryDto[]> {
+    return this.cryptoInputService.getHistory(jwt.id, +id);
+  }
```

Comment:
I think we should not return CryptoInputHistory here, but also BuyCryptoHistory. CryptoInputs are just inputs. Depending on the route, either a BuyFiat, a BuyCrypto or a CryptoStaking is created. So for CryptoRoute, you'll also find the history in the BuyCrypto table.

Reply:
okay, fine for me. It was more or less the only entry point i saw into crypto routes. But if another approach is more feasible and can gather more information, why not?